storage/client/src/test/java/com/azure/storage/blob/AppendBlobAPITest.groovy

```diff
@@ -0,0 +1,519 @@
+        null     | null       | null         | garbageEtag | null
+        null     | null       | null         | null        | receivedLeaseID
+    }
+
+    @Unroll
+    def "Create AC fail"() {
+        setup:
+        noneMatch = setupBlobMatchCondition(bu, noneMatch)
+        setupBlobLeaseCondition(bu, leaseID)
+        BlobAccessConditions bac = new BlobAccessConditions()
+            .leaseAccessConditions(new LeaseAccessConditions().leaseId(leaseID))
+            .modifiedAccessConditions(new ModifiedAccessConditions().ifModifiedSince(modified)
+                .ifUnmodifiedSince(unmodified)
+                .ifMatch(match)
+                .ifNoneMatch(noneMatch))
+
+        when:
+        bu.create(null, null, bac, null)
+
+        then:
+        thrown(StorageException)
+
+        where:
+        modified | unmodified | match       | noneMatch    | leaseID
+        newDate  | null       | null        | null         | null
+        null     | oldDate    | null        | null         | null
+        null     | null       | garbageEtag | null         | null
+        null     | null       | null        | receivedEtag | null
+        null     | null       | null        | null         | garbageLeaseID
+    }
+
+    /*def "Create context"() {
+        setup:
+        def pipeline = HttpPipeline.build(getStubFactory(getContextStubPolicy(201, AppendBlobCreateHeaders)))
+
+        bu = bu.withPipeline(pipeline)
+
+        when:
+        bu.create(null, null, null, defaultContext)
+
+        then:
+        notThrown(RuntimeException)
+    }*/
+
+    def "Append block defaults"() {
+        setup:
+        Response<AppendBlobItem> appendResponse = bu.appendBlock(defaultInputStream.get(), defaultDataSize)
+
+        expect:
+        ByteArrayOutputStream downloadStream = new ByteArrayOutputStream()
+        bu.download(downloadStream)
+
+        downloadStream.toByteArray() == defaultData.array()
+        validateBasicHeaders(appendResponse.headers())
+        appendResponse.value().contentMD5() != null
+        appendResponse.value().blobAppendOffset() != null
+        appendResponse.value().blobCommittedBlockCount() != null
+        Integer.parseInt(bu.getProperties().headers().value("x-ms-blob-committed-block-count")) == 1
+    }
+
+    def "Append block min"() {
+        expect:
+        bu.appendBlock(defaultInputStream.get(), defaultDataSize).statusCode() == 201
+    }
+
+    @Unroll
+    def "Append block IA"() {
+        when:
+        bu.appendBlock(data, dataSize)
+
+        then:
+        def e = thrown(Exception)
+        exceptionType.isInstance(e)
+
+        where:
+        data                        | dataSize            | exceptionType
+        null                        | defaultDataSize     | NullPointerException
+        defaultInputStream.get()    | defaultDataSize + 1 | IndexOutOfBoundsException
+        //defaultInputStream.get()    | defaultDataSize - 1 | StorageException
```

Comment:
That last test runs successfully as it will read all but the last byte of data from the input stream and send that off to the service.

Thoughts on allow this?

Reply:
I'm not sure what the original test was trying to trip. @rickle-msft would know best.

My guess is that the entire stream was flushed to the network, but we sent more than we said we would, and so the service rejected the call. It seems that is no longer the behavior, if this test is passing.

Reply:
Created an issue to track/discuss this https://github.com/Azure/azure-sdk-for-java/issues/4193. 

Context:

UnexpectedLengthException was an exception from v2 runtime, we don't have this exception in new preview runtime. In v2 this exception was thrown from our own implementation of netty-rx-adapter. For new runtime we use third party netty-reactor-layer, we should be able to apply an operator to the content flux before passing it to netty-reactor-layer.